java/src/main/java/ai/rapids/cudf/HostColumnVectorCore.java

```diff
@@ -115,29 +125,260 @@ Object getElement(int rowIndex) {
       int start = offHeap.offsets.getInt(rowIndex * DType.INT32.getSizeInBytes());
       int end = offHeap.offsets.getInt((rowIndex + 1) * DType.INT32.getSizeInBytes());
       for (int j = start; j < end; j++) {
-        retList.add(children.get(0).getElement(j));
+        for (HostColumnVectorCore childHcv : children) {
+          retList.add(childHcv.getElement(j));
+        }
       }
       return retList;
-    } else if (type == DType.STRING) {
-      int start = offHeap.offsets.getInt(rowIndex * DType.INT32.getSizeInBytes());
-      int end = offHeap.offsets.getInt((rowIndex + 1) * DType.INT32.getSizeInBytes());
-      int size = end - start;
-      byte[] rawData = new byte[size];
-      if (size > 0) {
-        offHeap.data.getBytes(rawData, 0, start, size);
-        return new String(rawData);
-      } else if (isNull(rowIndex)) {
+    } else if (type == DType.STRUCT) {
+      return getStruct(rowIndex);
+    } else {
+      if (isNull(rowIndex)) {
         return null;
-      } else {
-        return new String();
       }
+      return readValue(rowIndex);
+    }
+  }
+
+  private Object getString(int rowIndex) {
+    int start = offHeap.offsets.getInt(rowIndex * DType.INT32.getSizeInBytes());
+    int end = offHeap.offsets.getInt((rowIndex + 1) * DType.INT32.getSizeInBytes());
+    int size = end - start;
+    byte[] rawData = new byte[size];
+    if (size > 0) {
```

Comment:
A few nits in here. First if it is null it does not matter what the size is.  Technically a null can have data in it that is just masked.  I would prefer to see the null check the first thing, or possibly outside of the private get* methods just at the top.

Reply:
Moved `isNull` check